serverlessworkflow/modules/ROOT/pages/data-index/data-index-core-concepts.adoc

```diff
@@ -0,0 +1,635 @@
+
+image::data-index/data-index-dev-service.png[Image of data-index deployment an Quarkus Dev Service]
+
+More details are provided in the xref:data-index/data-index-service.adoc#data-index-dev-service-details[{data_index_ref} as a Quarkus Development Service] section.
+
+The {product_name} Process Quarkus extension sets up your Quarkus application to automatically replicate any {product_name} messaging events related to {workflow_instances} or jobs into the provisioned Data Index instance.
+
+For more information about Quarkus Dev Services, see link:{dev_services_url}[Dev Services guide].
+
+=== {data_index_ref} Quarkus extension added to the {product_name} application
+It can be included as part of the same {product_name} application using the *{data_index_ref} extension*, through the provided addons.
+This scenario is specific to add the {data_index_ref} audit data indexing features and the GraphQL endpoint exposure inside a workflow application.
+Even though it has some limitations regarding the audit data it can provide just from the single application where the extension is added it allows to simplify the communication between services regarding the indexation of the audit data is something internal to the application, with no extra communication configuration.
+
+image::data-index/data-index-addon.png[Image of data-index as a Quarkus Extension]
+
+More details are available in the xref:data-index/data-index-quarkus-extension.adoc[{data_index_ref} Quarkus Extension] section.
+
+[#data-index-storage]
+== Audit Data storage
+
+In order to store the indexed data, {data_index_ref} needs some specific tables to be created. {data_index_ref} is ready to use link:{flyway_quarkus_url}[Quarkus flyway] for that purpose.
+
+It's necessary to activate the migrate-at-start option to migrate the {data_index_ref} schema automatically.
+
+There are more explanations about flyway migrations related in xref:persistence/postgresql-flyway-migration.adoc[Migrating Your PostgreSQL Database with a Kogito upgrade] section
+
+[#data-index-graphql]
+== GraphQL endpoint provided by {data_index_ref}
+
+{data_index_ref} provides Graphql endpoint that allows users to interact with the stored audit data.
+For more information about GraphQL see {graphql_url}[GraphQL]
+
+[#data-index-ext-queries]
+=== GraphQL queries for {workflow_instances} and jobs
+
+Use the following GraphQL queries to retrieve data about {workflow_instance}s and jobs.
+
+Retrieve data from {workflow_instances}::
++
+--
+You can retrieve data about a specified {workflow_instance} from your Workflow definition.
+
+.Example query
+[source]
+----
+{
+  ProcessInstances {
+    id
+    processId
+    state
+    parentProcessInstanceId
+    rootProcessId
+    rootProcessInstanceId
+    variables
+    nodes {
+      id
+      name
+      type
+    }
+  }
+}
+----
+--
+
+Retrieve data from jobs::
```

Comment:
```suggestion
Retrieve data from jobs:
```

Reply:
I can't remove that, it's needed to keep the guide format.